src/edges_analysis/analysis/levels.py

```diff
@@ -1044,51 +1046,32 @@ def _promote(
             The filename of the ACQ file to read.
         band
             Defines the instrument that took the data (mid, low, high).
-        calfile
-            A file containing the output of
-            :meth:`edges_cal.CalibrationObservation.write` -- i.e. all the information
-            required to calibrate the raw data. Determination of calibration parameters
-            occurs externally and saved to this file.
-        s11_path
-            Path to the receiver S11 information relevant to this observation.
         weather_file
             A weather file to use in order to capture that information (may find the
             default weather file automatically).
         thermlog_file
             A thermlog file to use in order to capture that information (may find the
             default thermlog file automatically).
-        leave_progress
-            Whether to leave the progress bar on the screen at the end.
-        xrfi_pipe
-            A dictionary in which keys specify xrfi method names (see
-            :mod:`edges_cal.xrfi`) and values are dictionaries which specify the
-            parameters to be passed to those methods (not requiring the spectrum/weights
-            arguments).
-        s11_file_pattern
-            A format string defining the naming pattern of S11 files at ``s11_path``.
-            This is used to automatically find the S11 file closest in time to the
-            observation, if the ``s11_path`` is not explicit (i.e. it is a directory).
-        ignore_s11_files
-            A list of paths to ignore when attempting to find the S11 file closest to
-            the observation (perhaps they are known to be bad).
-
-        Other Parameters
-        ----------------
-        All other parameters are passed to :meth:`_calibrate` -- see its documentation
-        for details.
-
-        Returns
-        -------
-        data
-            An instantiated :class:`CalibratedData` object.
         """
         t = time.time()
-        q = prev_step["spectra"]["Q"]
+        freq_f = prev_step["freq_ancillary"]["frequencies"]
+        f_mask = np.where(freq_f.freq >= 40)[0]
+        freq = freq_f[f_mask:]
+        q = prev_step["spectra"]["Q"][
+            f_mask:,
+        ]
```

Comment:
I think a quicker/neater way of doing the masking here would be to use the `FrequencyRange` object:

```python
freq_rng = FrequencyRange(prev_step['freq_ancillary']['frequencies'], f_low=40.0)
freq = freq_rng.freq
q = prev_step['spectra']['Q'][freq_rng.mask]
...
```